### PR TITLE
[DTensor] implement aten.equal sharding prop

### PR DESCRIPTION
**Pull request by XilunWu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #95852
* __->__ #97170

### CODE REVIEW

**Review by wanchaol — APPROVED** [review]
lgtm, thanks for the fix!

File: test/distributed/_tensor/test_tensor_ops.py [hunk 1]
```diff
@@ -234,6 +234,27 @@ def test_zeros_like(self):
...
+        if self.rank == 0:
+            input_tensor_1 = torch.ones(4, 4)
+        else:
+            input_tensor_1 = torch.randn(4, 4)
+        input_tensor_2 = torch.ones(4, 4)
+
+        dist_tensor_1 = DTensor.from_local(input_tensor_1, device_mesh, shard_spec)
+        dist_tensor_2 = DTensor.from_local(input_tensor_2, device_mesh, shard_spec)
+        # TODO: equal op currently returns each shard's local result
+        eq_result = dist_tensor_1.equal(dist_tensor_2)
```
Comment (wanchaol) [inline test/distributed/_tensor/test_tensor_ops.py:251]: maybe this should be under a raise Exception condition? It's wrong now, so we would fix it later by removing that exception

**Comment by XilunWu** [comment]
@XilunWu has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44321773).

**Comment by XilunWu** [comment]
@XilunWu has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44321773).

**Pull request closed without merging**